persistence/gov.go

```diff
@@ -908,38 +109,85 @@ func (p PostgresContext) SetParam(paramName string, paramValue interface{}) erro
 	if err != nil {
 		return err
 	}
-	if _, err = tx.Exec(ctx, schema.NullifyParamsQuery(height)); err != nil {
-		return err
-	}
-	if _, err = tx.Exec(ctx, schema.SetParam(paramName, paramValue, height)); err != nil {
+	if _, err = tx.Exec(ctx, schema.SetParamQuery(paramName, paramValue, height)); err != nil {
 		return err
 	}
 	return tx.Commit(ctx)
 }
 
-func (p PostgresContext) GetIntParam(paramName string) (i int, err error) {
+func GetParam[T int | string | []byte](p PostgresContext, paramName string, height int64) (i T, err error) {
 	ctx, conn, err := p.GetCtxAndConnection()
 	if err != nil {
-		return 0, err
+		return i, err
+	}
+
+	var stringVal string
+	err = conn.QueryRow(ctx, schema.GetParamQuery(paramName, height)).Scan(&stringVal)
+	switch tp := any(i).(type) {
+	case int, int32, int64:
+		iConv, errr := strconv.Atoi(stringVal)
+		if errr != nil {
+			err = errr
+			return
+		}
+		return any(iConv).(T), err
+	case string:
+		return any(stringVal).(T), err
+	case []byte:
+		v, e := hex.DecodeString(stringVal)
+		return any(v).(T), e
+
+	default:
+		log.Fatalf("unhandled type for paramValue %T", tp)
 	}
-	err = conn.QueryRow(ctx, schema.GetParamQuery(paramName)).Scan(&i)
+
 	return
 }
 
-func (p PostgresContext) GetStringParam(paramName string) (s string, err error) {
+func SetFlag[T schema.SupportedParamTypes](p PostgresContext, paramName string, paramValue T, enabled bool) error {
 	ctx, conn, err := p.GetCtxAndConnection()
 	if err != nil {
-		return "", err
+		return err
 	}
-	err = conn.QueryRow(ctx, schema.GetParamQuery(paramName)).Scan(&s)
-	return
+	height, err := p.GetHeight()
+	if err != nil {
+		return err
+	}
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{})
+	if err != nil {
+		return err
+	}
+	if _, err = tx.Exec(ctx, schema.SetFlagQuery(paramName, paramValue, enabled, height)); err != nil {
+		return err
+	}
+	return tx.Commit(ctx)
 }
 
-func (p PostgresContext) GetBytesParam(paramName string) (param []byte, err error) {
+func GetFlag[T int | string | []byte](p PostgresContext, paramName string, height int64) (i T, enabled bool, err error) {
```

Comment:
I definitely think that the entrypoints to setting / getting the Flag / Param should definitely be different for code readability purposes, but since most of the code is the same, I think you can create a helper that takes in a function for the query creation to reduce the code footprint.

See here for an example of how we did it elsewhere: https://github.com/pokt-network/pocket/blob/main/persistence/account.go#L143

Reply:
I had a look and even if I like the approach, I feel like in here it can be simplified while still keeping it DRY.

Reply:
The final result LGTM